homeassistant/components/bluetooth/passive_update_processor.py

```diff
@@ -35,22 +42,69 @@ class PassiveBluetoothEntityKey:
 
 _T = TypeVar("_T")
 
+PASSIVE_UPDATE_PROCESSOR_COORDINATORS = "passive_update_processor_coordinators"
+
 
 @dataclasses.dataclass(slots=True, frozen=True)
 class PassiveBluetoothDataUpdate(Generic[_T]):
     """Generic bluetooth data."""
 
     devices: dict[str | None, DeviceInfo] = dataclasses.field(default_factory=dict)
-    entity_descriptions: Mapping[
+    entity_descriptions: dict[
         PassiveBluetoothEntityKey, EntityDescription
     ] = dataclasses.field(default_factory=dict)
-    entity_names: Mapping[PassiveBluetoothEntityKey, str | None] = dataclasses.field(
+    entity_names: dict[PassiveBluetoothEntityKey, str | None] = dataclasses.field(
         default_factory=dict
     )
-    entity_data: Mapping[PassiveBluetoothEntityKey, _T] = dataclasses.field(
+    entity_data: dict[PassiveBluetoothEntityKey, _T] = dataclasses.field(
         default_factory=dict
     )
 
+    def async_restore_data(self) -> dict[str, Any]:
+        return {
+            "devices": self.devices,
+            "entity_descriptions": self.entity_descriptions,
+            "entity_names": self.entity_names,
+            "entity_data": self.entity_data,
+        }
+
+
+def register_coordinator(
+    hass: HomeAssistant, coordinator: PassiveBluetoothProcessorCoordinator
+) -> None:
+    """Register a coordinator to have its processors data restored."""
+    coordinators: set[PassiveBluetoothProcessorCoordinator] = hass.data[
+        PASSIVE_UPDATE_PROCESSOR_COORDINATORS
+    ]
+    coordinators.add(coordinator)
+
+
+def unregister_coordinator(
+    hass: HomeAssistant, coordinator: PassiveBluetoothProcessorCoordinator
+) -> None:
+    """Unregister a coordinator to have its processors data restored."""
+    coordinators: set[PassiveBluetoothProcessorCoordinator] = hass.data[
+        PASSIVE_UPDATE_PROCESSOR_COORDINATORS
+    ]
+    coordinators.remove(coordinator)
+
+
+async def async_setup(hass: HomeAssistant) -> None:
+    """Setup the passive update processor coordinators."""
+    coordinators: set[PassiveBluetoothProcessorCoordinator] = set()
+    hass.data[PASSIVE_UPDATE_PROCESSOR_COORDINATORS]: set[
+        PassiveBluetoothProcessorCoordinator
+    ] = coordinators
+
```

Comment:
Need to make storage here as well.  Probably s new object that has storage hanging of it